hibernate-core/src/main/java/org/hibernate/query/sqm/produce/function/internal/PatternRenderer.java

```diff
@@ -71,10 +72,18 @@ else if ( c  == '?' ) {
 					}
 				}
 
-				Integer paramNumber = Integer.valueOf( index.toString() );
-				paramNumbers.add( paramNumber );
-				paramList.add( paramNumber );
-				index.setLength(0);
+				if ( index.toString().endsWith("...") ) {
```

Comment:
I think the above code might incur some performance cost by creating a new string to test. Maybe we can use `StringBuilder.lastIndexOf()` to avoid that. If such processing is common enough, we can create a util method in StringHelper.